install

```diff
@@ -13,69 +19,125 @@ usage() {
   exit 1
 }
 
-INSTALLED_VIA_HOMEBREW=false
-
+# Reads all options passed in and sets environment variables or shows usage/help
 while getopts ":pub" opt; do
   case $opt in
     p ) SKIP_SETTING_PATH=1;;
-    u ) DO_UNINSTALL=1;;
     b ) INSTALLED_VIA_HOMEBREW=true;;
     * ) usage ;;
   esac
 done
 
-if [ -d "$FASTLANE_DIR" ]; then
-  echoc "fastlane is already installed at $FASTLANE_DIR" yellow
-  if [ "$DO_UNINSTALL" == "1" ]; then
-    choice="y"
+install_or_upgrade_via_homebrew()
+{
+  PACKAGE=$1
+  list_output=`brew list | grep $PACKAGE`
+  outdated_output=`brew outdated | grep $PACKAGE`
+
+  if [[ ! -z "$list_output" ]]; then
+    echo "$PACKAGE is installed"
+    if [[ -z "$outdated_output" ]]; then
+      echo "$PACKAGE is updated"
+    else
+      echo "$PACKAGE is not updated"
+      prompt_to_install_or_upgrade $PACKAGE
+      brew upgrade $PACKAGE
+    fi
   else
-    read -p "Do you want to reinstall fastlane? The old directory, including any changes you made, will be removed! (y/N) " -n 1 choice
-    echo ""
+    echo "$PACKAGE is not installed."
+    prompt_to_install_or_upgrade $PACKAGE
+    brew install $PACKAGE
   fi
+}
 
-  case $choice in
-    y|Y )
-      echoc "The installer will now run the uninstall script to remove your existing fastlane bundle..." magenta
-
-      if [ "$DO_UNINSTALL" == "1" ]; then
-        "$CURRENT_DIR/uninstall" -y
-      else
-        "$CURRENT_DIR/uninstall"
-      fi
-
-      if [[ $? != 0 ]]; then
-        echoc "There was a problem uninstalling the previous fastlane bundle, aborting installation..." red
-        exit 1
-      fi
-      echoc "Done uninstalling, your regular installation will now commence. 🚕 💨" green
-      ;;
-    * )
-      echoc "Installation aborted, nothing to be done!" green
-      exit 1
-      ;;
+prompt_to_install_or_upgrade()
+{
+  PACKAGE=$1
+
+  read -p "The installer needs to install/upgrade $PACKAGE with brew. Is that okay? (y/N) " -n 1 choice
```

Comment:
It would be very helpful to skip the prompts with a the help of a command line option. Automating the installation a little bit easier.